File: spring/pcbuilder/src/main/java/ar/edu/um/pcbuilder/dtos/entities/devices/DeviceDto.java
package ar.edu.um.pcbuilder.dtos.entities.devices;

import java.util.List;
import lombok.Data;

@Data
public class DeviceDto {

  private Long id;
  private String code;
  private String name;
  private String description;
  private Double basePrice;
  private String currency;
  private List<DeviceCharacteristicDto> characteristics;
  private List<DevicePersonalizationDto> personalizations;
  private List<DeviceOptionDto> options;
  private List<DeviceAdditionalItemDto> additionalItems;
}

File: kmp/composeApp/src/commonMain/kotlin/ar/edu/um/pcbuilder/dtos/entities/devices/DeviceDto.kt
package ar.edu.um.pcbuilder.dtos.entities.devices

data class DeviceDto(
    val id: Long?,
    val code: String?,
    val name: String?,
    val description: String?,
    val basePrice: Double?,
    val currency: String?,
    val characteristics: List<DeviceCharacteristicDto>?,
    val personalizations: List<DevicePersonalizationDto>?,
    val options: List<DeviceOptionDto>?,
    val additionalItems: List<DeviceAdditionalItemDto>?
)